### ISSUE TITLE
Can not write text in text file using text editor view.

### ISSUE THREAD

**Issue by afrinj** [redhat-developer/vscode-extension-tester#329]
source : https://github.com/redhat-developer/vscode-extension-tester/blob/master/test/test-project/src/test/editor/textEditor-test.ts

Modified code:

```
    import { expect } from 'chai';
    import { TextEditor, EditorView, StatusBar, InputBox, Workbench } from "vscode-extension-tester";

    describe('TextEditor', () => {
        let editor: TextEditor;
        let editor1: TextEditor;
        let view: EditorView;

        const testText = process.platform === 'win32' ? `line1\r\nline2\r\nline3` : `line1\nline2\nline3`;

        before(async function() {
            this.timeout(8000);
            await new Workbench().executeCommand('File: New File');
            await new Promise((res) => { setTimeout(res, 1000); });
            view = new EditorView();
            editor = new TextEditor(view);
            await new Promise((res) => { setTimeout(res, 1000); });
      
         await new StatusBar().openLanguageSelection();
        const input = await InputBox.create();
        await input.setText('typescript');
        await input.confirm();


    });

    after(async () => {
        await editor.clearText();
        await view.closeAllEditors();
    });

    it('can get and set text', async () => {
        await editor.setText(testText);
        const text = await editor.getText();
        expect(text).equals(testText);
    });

    it('can get and set text at line', async () => {
        await editor.setTextAtLine(1, 'line1');
        const line = await editor.getTextAtLine(1);
        expect(line).has.string('line1');
    });

    it('can type text at given coordinates', async function() {
        this.timeout(5000);
        await editor.typeText(1, 5, '1222222');
        const line = await editor.getTextAtLine(1);
        expect(line).has.string('line11');
    });
});

```

All of those are failing:

```
0 passing (11s)
  3 failing

  1) TextEditor
       can get and set text:

      AssertionError: expected '' to equal 'line1\nline2\nline3'
      + expected - actual

      +line1
      +line2
      +line3
      
      at /home//-two/tool-suite/vscode-ide/packages/-vscode-ide/out/test/ui/component/Home.test.js:40:29
      at Generator.next (<anonymous>)
      at fulfilled (out/test/ui/component/Home.test.js:5:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  2) TextEditor
       can get and set text at line:
     AssertionError: expected '' to contain 'line1'
      at /home//-two/tool-suite/vscode-ide/packages/-vscode-ide/out/test/ui/component/Home.test.js:45:33
      at Generator.next (<anonymous>)
      at fulfilled (out/test/ui/component/Home.test.js:5:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  3) TextEditor
       can type text at given coordinates:
     AssertionError: expected '' to contain 'line11'
      at Context.<anonymous> (out/test/ui/component/Home.test.js:52:37)
      at Generator.next (<anonymous>)
      at fulfilled (out/test/ui/component/Home.test.js:5:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

Actually i was trying to set value in my json file by this same way. 

Any suggestion?

Thanks a lot

**Comment by jrichter1** [comment]
I'm not sure what happened here. There is no apparent reason why the editor should fail to set the text properly.

I even copied this code and ran it without problems, except the last test now expects a value that is not there.
```
  1) TextEditor
       can type text at given coordinates:
     AssertionError: expected 'line12222221' to contain 'line11'
      at Context.<anonymous> (out/src/test/editor/foo-test.js:52:37)
      at Generator.next (<anonymous>)
      at fulfilled (out/src/test/editor/foo-test.js:5:58)
      at processTicksAndRejections (node:internal/process/task_queues:94:5)
```

**Comment by odockal** [comment]
Is this issue still actual? Is it something we should take a look or did you manage to find a solution? Thanks.

**Comment by hall** [comment]
I get this error with the following test:

```javascript
import { expect } from 'chai';
import { TextEditor, VSBrowser } from 'vscode-extension-tester';

describe('Extension', () => {
    it('works', async () => {
        await VSBrowser.instance.openResources('test.md');
        const editor = await new TextEditor();
        await editor.setText("test");
        expect(await editor.getText()).equals("test");
    });
});
```

It seems `setText` doesn't write anything to the file; it does, however, open the content assist element (see the screenshot below from `./test-resources/screenshots/Extension works.png`).
If I swap `setText` for `typeText`, the text _is_ correctly written to the file but the test still fails with the same error.

![Extension works](https://user-images.githubusercontent.com/32495955/172027161-54de862b-e956-413c-a41e-4d2126fba537.png)

In fact, it appears `getText()` is returning the contents of the clipboard; e.g., if I first copy line 9 from the snippet above, I get the following error:

```
Launching browser...
Browser ready in 3484 ms
Launching tests...
  Extension
    1) works

Shutting down the browser

  0 passing (6s)
  1 failing

  1) Extension
       works:

      AssertionError: expected '        expect(await editor.getText()…' to equal 'test'
      + expected - actual

      -        expect(await editor.getText()).equals("test");
      +test
      
      at Context.<anonymous> (out/test/extension.test.js:10:52)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

This is on the latest version:

```
❯ jq -r '.packages."node_modules/vscode-extension-tester".version' package-lock.json
4.2.5
```

**Comment by hall** [comment]
Not sure _why_ this is the case but I think it might be the underlying problem: [the `textarea` being selected](https://github.com/redhat-developer/vscode-extension-tester/blob/71436dba8c9bb1e755d99d061da32deb622b666c/page-objects/src/components/editor/TextEditor.ts#L100) always has an empty string `value`.

![image](https://user-images.githubusercontent.com/32495955/172034330-73c869ac-a2e4-419d-a6f5-959ba4f83005.png)

I get this behavior in both Linux (VSCodium and a stock VSCode install) and Windows (a fresh VM, using the official installer), all on version 1.67.2.

However, the  value _is_ set in https://vscode.dev (which is also at version 1.67.2 and I tested in Firefox, Chromium, and Chrome).

![image](https://user-images.githubusercontent.com/32495955/172034441-a95e8633-eed7-4d43-8b38-914401800ceb.png)

So this is probably an upstream bug. Hopefully others more familiar with the app can reproduce this behavior.

**Comment by hall** [comment]
I figured it out! (and I feel silly b/c I definitely tested this early on but turns out I did the test wrong :upside_down_face: )

It works if I change [the `ctrl-a`, `ctrl-c` sequence of `getText`](https://github.com/redhat-developer/vscode-extension-tester/blob/71436dba8c9bb1e755d99d061da32deb622b666c/page-objects/src/components/editor/TextEditor.ts#L101) to `ctrl-a`, `ctrl-j` because I'm using the dvorak keyboard layout where qwerty `c` maps to dvorak `j`.

This is related to https://github.com/redhat-developer/vscode-extension-tester/issues/13 which I don't think is adequately solved. I'm not sure of a better solution at this point but I'll try to do some more digging to see what I can find.

**Comment by hall** [comment]
Alright, I've done about all the searching I care to do at this point. Seems like this is a chromedriver issue; there are [reports in their bug tracker](https://bugs.chromium.org/p/chromium/issues/detail?id=1295243) but they should be solved in the tested version (`98.0.4758.102` -- I even tried several previous versions to no avail; as far back as `91.0.4472.101` since there were claims it was a regression from `97` -> `98`).

Workaround for now is to set US qwerty as the _default_ layout (not just the active one) in OS preferences.
